x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -252,11 +249,39 @@ public IndexMetadata getSnapshotIndexMetaData(RepositoryData repositoryData, Sna
         return imdBuilder.build();
     }
 
+    /**
+     * Dummy request key for deduplicating all remote cluster state requests via {@link #getRemoteStateDeduplicator}.
+     */
+    private static final Object RESULT_KEY = new Object();
```

Comment:
This is a little awkward but I think it's good enough and I didn't want to build a whole new things for the deduplication here when the existing deduplicator works just fine for what we need here this way ...